build-test.cmd

```diff
@@ -6,8 +6,17 @@ set __BuildArch=x64
 set __VCBuildArch=x86_amd64
 set __BuildType=Debug
 set __BuildOS=Windows_NT
-set __VSVersion=vs2015
-set __VSToolsRoot=%VS140COMNTOOLS%
+
+:: Default to highest Visual Studio version available
+if defined VS150COMNTOOLS (
```

Comment:
Running from a regular command prompt depends on the VS***COMNTOOLS environment variable to be defined.

For VS2015 and prior, this is a global environment variable set by the vs installer. For VS2017 and later, there is no global environment variable and no registry keys. In order to get VS150COMNTOOLS you have to run developer command prompt.

Launching vcvarsall.bat itself is dependent on us being in developer command prompt as well (so we can locate the script).

The logic I have here is: are you in the vs2017 command prompt?

If yes, assume vs 2017 tools.

Otherwise, use vs2015 tools.